### PR TITLE
Add RPM Packaging

### PR DESCRIPTION
**Pull request by djw8605**
Here is the initial pull request.  Do not approve until I have:

- [x] Written documentation on how to override attributes in the README.md
- [x] Tested on the hcc-cvmfs-repo node.

### CODE REVIEW

**Comment by bbockelm** [comment]
@djw8605 - what's the status here?

**Comment by bbockelm** [comment]
Update: chatted with Derek on this.  Still needs to be tested on the CVMFS repo host.  Hoping to do that next week (after the OSG summer school).

**Comment by djw8605** [comment]
@bbockelm I have some questions.

1. What is the reasoning behind putting ligo_authz in the $PWD?  Is it a security risk to put it in /tmp?
2. Is the ligo "whitelist" updated frequently?  Should that be included in the RPM?

...
I'll probably come up with something more later.

**Comment by bbockelm** [comment]
- `ligo_authz` is a well-known filename, meaning that it can't go in`/tmp` without opening up security issues.
- At this point, whitelist should live in Puppet, not the RPM.

**Comment by djw8605** [comment]
Does it have to be named `ligo_authz`?  I think it's regenerated each iteration.

**Comment by bbockelm** [comment]
It doesn't have to be named `ligo_authz` -- if you generate a filename with `mkstemp` (and don't pollute the directory full of scratch files!), then using `/tmp` should be fine.

**Comment by bbockelm** [comment]
@djw8605 - ready to go?  I have a pretty big incoming PR in the `cms-sync-v2` branch and you will be happier if this branch lands first.

**Comment by djw8605** [comment]
I just committed the temporary authz file fix.  This is so that the `/usr/share/cvmfs-sync` directory (the working directory for cvmfs-sync services) doesn't have to be writable by the `bbockelm` (or whatever repository owner) user.

Look over the changes in the commits.  It's not a perfect situation, since it will likely fail if the `ligo-auth-gen` script fails.

File: update-scripts/ligo-data-update [hunk 1]
```diff
@@ -34,9 +37,11 @@ else
...
 
 # Generate a new authz list which we will use for publication
-if `./ligo-auth-gen ligo_authz -w whitelist`; then
+tmp_file=`mktemp`
+if `/usr/libexec/cvmfs-sync/ligo-auth-gen $tmp_file -w whitelist`; then
   echo "Successfully updated LIGO user list."
   query_failed=0
+  authz_location=$tmp_file
 else
   echo "Failed to update LIGO user list but proceeding with update."
```
Comment (bbockelm) [inline update-scripts/ligo-data-update]: Just go ahead and fail the update -- if the LIGO user list fails, then you have the potential of locking yourself out!

**Comment by djw8605** [comment]
@bbockelm ok, added code to abort update if the auth query fails.

**Comment by bbockelm** [comment]
For completeness - is the second checkbox done?

**Comment by djw8605** [comment]
Ah, yes.  hcc-cvmfs-repo is now running the RPM.  I need to update it with the most recent ligo changes, but I am doing that now.

**Comment by bbockelm** [comment]
Ok, let's roll!  With this done, we can start working to merge the CMS branch.

**Pull request merged**